File: 백준/Silver/10828. 스택/스택.py
import sys
T = int(input())

stack = []
size = 0
for i in range(1, T + 1):
    order = sys.stdin.readline().rstrip()
    if order == 'pop':
        if stack:
            print(stack.pop())
            size -= 1
        else:
            print(-1)
    elif order == 'size':
        print(size)
    elif order == 'empty':
        if size == 0:
            print(1)
        else:
            print(0)
    elif order == 'top':
        if stack:
            print(stack[-1])
        else:
            print(-1)
    else:
        stack.append(int(order[5:]))
        size += 1
    if i == T:
        break

File: 백준/Silver/10828. 스택/스택.cc
#include <bits/stdc++.h>

using namespace std;

int main()
{
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int n;
    cin >> n;
    stack<int> stack;
    while(n--)
    {
        string s;
        cin >> s;

        if (s == "push")
        {
            int t;
            cin >> t;
            stack.push(t);
        }

        if (s == "pop" && !stack.empty())
        {
            cout << stack.top() << '\n';
            stack.pop();
        }
        else if (s == "pop") cout << -1 << '\n';

        if (s == "size") cout << stack.size() << '\n';

        if (s == "empty") cout << (stack.empty() ? 1 : 0) << '\n';
        
        if (!stack.empty() && s == "top") cout << stack.top() << '\n';
        else if (s == "top") cout << -1 << '\n';
    }
}
